### PR TITLE
Showing a spinner when drawing the random person

### LINKED ISSUE #21: Admin only: show a gif and delay showing the drawed person a bit
When admin clicks on a draw button we should show a gif like:

Feel free to pick other better gif, this is the first I found.

Also delay a little bit before showing the person just so the gif can play a bit then show the name and hide the gif

### PR DESCRIPTION
**Pull request by dhonysilva**
Close #21 

Here is the result of the spinner implementation:

https://github.com/lubien/sorteios/assets/179104/5f219cb6-27ee-4dea-99d3-ee377d08a6aa

The spinner stays for 5 seconds while waiting for the next random person being selected.

It can be improved on the future using a better animation and also apply some kind of shaky effect when the new random person appears. But for now, it should be helpful.

### CODE REVIEW

File: lib/sorteios_web/live/room_live/show.ex [hunk 1]
```diff
@@ -28,20 +28,28 @@ defmodule SorteiosWeb.RoomLive.Show do
         |> EQRCode.encode()
         |> EQRCode.svg(width: 240)
 
-      {:ok,
```
Comment (lubien) [inline lib/sorteios_web/live/room_live/show.ex:31]: I dont think this change is needed?

File: lib/sorteios_web/live/room_live/show.ex [hunk 2]
```diff
@@ -95,17 +103,41 @@ defmodule SorteiosWeb.RoomLive.Show do
     end
   end
 
+
   def handle_event("pick_a_random_person", _params, socket) do
+
+    send(self(), :run_search)
```
Comment (lubien) [inline lib/sorteios_web/live/room_live/show.ex]: Prefer Process.send_after. 3 seconds should be enough!

File: lib/sorteios_web/live/room_live/show.ex [hunk 3]
```diff
@@ -95,17 +103,41 @@ defmodule SorteiosWeb.RoomLive.Show do
...
+
+    socket =
+      assign(
+        socket,
+        randon_person: [],
+        loading: true
```
Comment (lubien) [inline lib/sorteios_web/live/room_live/show.ex]: Can we rename this assign to `loading_winner?` (including the `?` too)

File: lib/sorteios_web/live/room_live/show.ex [hunk 4]
```diff
@@ -95,17 +103,41 @@ defmodule SorteiosWeb.RoomLive.Show do
...
+      assign(
+        socket,
+        random_person: random_person,
+        loading: false
+      )
+
+    {:noreply, socket}
   end
 
+
```
Comment (lubien) [inline lib/sorteios_web/live/room_live/show.ex]: Remove blanks

File: lib/sorteios_web/live/room_live/show.html.heex [hunk 5]
```diff
@@ -225,14 +230,16 @@
...
               <%= if @random_person do %>
-                <.user_block user={@random_person} show_email?={@admin?} />
-                <button
-                  phx-click="give_prize_to_random_person"
-                  data-confirm="Are you sure? This cannot be undone"
-                  class="block w-full items-center justify-center rounded-md border border-transparent bg-indigo-600 px-4 py-2 font-medium text-white shadow-sm hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2 sm:my-2 sm:w-auto sm:text-sm"
-                >
-                  Give them the next prize
-                </button>
+                <%= if @loading == false do %>
```
Comment (lubien) [inline lib/sorteios_web/live/room_live/show.html.heex]: you dont need a new nested if, change line 227 to `if @random_person && !@loading_winner? do`

File: mix.lock
Comment (lubien) [inline mix.lock:1]: Remove this file from changes please

File: lib/sorteios_web/templates/layout/root.html.heex [hunk 6]
```diff
@@ -7,6 +7,7 @@
     <meta name="csrf-token" content={csrf_token_value()} />
     <%= live_title_tag(assigns[:page_title] || "Your own prize drawing app", suffix: " · Sorteios") %>
     <link phx-track-static rel="stylesheet" href={Routes.static_path(@conn, "/assets/app.css")} />
+    <link phx-track-static rel="stylesheet" href={Routes.static_path(@conn, "/assets/style.css")} />
```
Comment (lubien) [inline lib/sorteios_web/templates/layout/root.html.heex]: Wait what?

**Review by lubien — CHANGES_REQUESTED** [review]
Awesome! A few comments

File: lib/sorteios_web/live/room_live/show.ex (hunk 1 above)
Comment (dhonysilva) [reply lib/sorteios_web/live/room_live/show.ex:31]: Somehow the `|> assign(:loading?, false)` wasn't working.

File: mix.lock
Comment (lubien) [inline mix.lock:1]: Can you recover this file? The goal is for no changes to mix.lock to appear here on this PR

**Review by lubien — CHANGES_REQUESTED** [review]
Almost!

**Review by lubien — APPROVED** [review]

**Pull request merged**